internal/controller/progressive/progressive.go

```diff
@@ -154,6 +158,22 @@ func ProcessResource(
 		return false, 0, err
 	}
 
+	// get UpgradingChildStatus and reset it if necessary
+	childStatus, err := getUpgradingChildStatus(ctx, rolloutObject, currentUpgradingChildDef)
+	if err != nil {
+		return false, 0, err
+	}
+
+	// if the Upgrading child status exists but indicates that we aren't done with upgrade process, then do postupgrade process
+	if !childStatus.InitializationComplete {
+		needsRequeue, err := startPostUpgradeProcess(ctx, rolloutObject, existingPromotedChild, currentUpgradingChildDef, controller, c)
+		if needsRequeue {
+			return false, common.DefaultRequeueDelay, err
+		} else {
+			return false, 0, err
+		}
+	}
```

Comment:
see that `startUpgradeProcess()` is called above if Upgrading child doesn't exist, and `startPostUpgradeProcess()` called here if `InitializationComplete==false`